### PR TITLE
[PSR-7] Added section on file uploads

### PR DESCRIPTION
**Pull request by weierophinney**
This adds a section on why `getFileParams()` is tied to the `$_FILES` superglobal, and the vision for how to handle file uploads by consumers.

See #492 for more information

### CODE REVIEW

**Comment by weierophinney** [comment]
Pinging @simensen to review.

File: proposed/http-message-meta.md [hunk 1]
```diff
@@ -613,6 +613,75 @@ Examples of this practice already exist in libraries such as
...
+as each specific file detail (e.g., content type) is nested with those of any
+sibling uploads.
+
+We considered specifying that `getFileParams()` MUST parse the values into a
+nested structure that mimics how they were provided in the form; however, we
+feel:
+
+- It introduces inconsistency in the API. Other parameters, such as query string
+  and cookie parameters, are specified such that they MUST retain the
+  structure of their corresponding superglobals. (Body parameters are a
```
Comment (webmozart) [inline proposed/http-message-meta.md]: I disagree with this statement. While the `$_POST` and `$_GET` arrays are usable by default, the `$_FILES` array isn't. Being consistent with `$_FILES` is not a desirable goal for me.

Two major PHP frameworks, i.e. ZF and Symfony, fix the `$_FILES` array in their implementations. Probably more. I think that it is irresponsible to ignore this issue in the PSR.

Comment (weierophinney) [inline proposed/http-message-meta.md]: In ZF, we don't handle it at the request level, though. We handle it in our `Zend\File\Transfer ` component, and the related filters and validators. Those are fed the raw `$_FILES` structure, which the request object encapsulates. (I've actually been blissfully unaware of how we do this until investigating for this issue.)

If anything, that validates the approach suggested here.

**Comment by weierophinney** [comment]
Closing in favor of #492.

**Pull request closed without merging**